### DISCUSSION TITLE
http://mburaksayici.com/blog/2024/09/21/easy-to-use-nosql-prompt-database-for-small-projects.html

### DISCUSSION THREAD

**Discussion by mburaksayici** [msiemens/tinydb#567] [Show and tell]
I needed a DB for my LLM project that:
- project isn’t really big
- don’t have to expose your prompts to business team, and even you are not really changing frequently.
- Have 100s of prompts
- don’t really do promp chaining/versioning/branching 
- use agents but don’t play with prompts to much
- just commit to git/bit to deploy, ie your CI/CD is simple as that, not complex
- mostly read from DB (you do actually)

I had used sqlite for one of the projects and has also seen that ChromaDB stores those vectors just in mssql. No external dependency, no need to deploy/make it up, perfect for small cases. Then I realise I wanted a database that:

SQLite for NoSQL, since we deal with prompts
No need to deploy and no need to add layer like having mongodb in docker-compose
Still I want to change prompts easily and commit to git/bit (Compare it with the case of MongoUI that you need to deploy the Mongo container, get into the MongoUI and change prompts)
Can organise prompts easily

And I quite liked the TinyDB. 

http://mburaksayici.com/blog/2024/09/21/easy-to-use-nosql-prompt-database-for-small-projects.html

Any opinions or contributions?

**Comment by msiemens** [comment]
Great article, thanks for sharing!
